src/app/tabs/compileTab/compilerContainer.js

```diff
@@ -113,6 +116,39 @@ class CompilerContainer {
     return el
   }
 
+  _setCompilerVersionFromPragma (filename) {
+    this.compileTabLogic.fileManager.getFile(filename).then(data => {
```

Comment:
It feel very "manual" couldn't we use the AST to get the information instead of manipulating strings ?

Reply:
+1 for AST and I would suggest using semver we already use it in the project

Reply:
okay, will do that

Reply:
Thanks fro semver suggestion @LianaHus , Now it works in less than half code